src/Components/SearchBar.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  useState,
+  useEffect,
+  SetStateAction,
+} from "react";
+import styled from "styled-components";
+import mediaQuery from "../static/MediaQuery";
+import { CardData } from "../Pages/Shared/type";
+import search from "../images/search.svg";
+import SearchClose from "../images/SearchClose.svg";
+
+const SearchBarContainer = styled.div`
+  margin: 40px auto;
+  width: 1060px;
+  position: relative;
+  ${mediaQuery.tablet} {
+    padding-left: 3.2rem;
+    padding-right: 3.2rem;
+    width: auto;
+  }
+`;
+const SearchImg = styled.img`
+  position: absolute;
+  left: 15px;
+  top: 15px;
+  ${mediaQuery.tablet} {
+    left: 4.4rem;
+  }
+`;
+const SearchCloseImg = styled.img`
+  position: absolute;
+  right: 15px;
+  top: 15px;
+  &:hover {
+    cursor: pointer;
+  }
+`;
+const SearchInput = styled.input`
+  width: 100%;
+  padding: 1.5rem 4rem 1.6rem;
+  background: var(--Grey-Light);
+  border-radius: 10px;
+  border: none;
+  outline: none;
+  appearance: none;
+
+  :focus {
+    outline: none;
+  }
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+interface Props {
+  linkData: { data: CardData[] };
```

Comment:
의도가 있는게 아니라면 데이터 구조의 중첩 단계를 줄이는 게 좋겠습니다.  
```suggestion
  linkData: CardData[];
```